### PR TITLE
Backend testing

### PR DESCRIPTION
**Pull request by celdond**
Took far more effort than expected, and I had to fix a small but important error.

Testing suite has 80% statement and line coverage now.
I documented out the conventions being used in the tests files as of right now.
There are still some things that need to be done:
- WebSocket Messages need to be checked
- Documentation on each file's page goals
- Clean-up organization

The major fix that needs to be in main ASAP is the alteration in game.ts from WebSocket.OPEN to someSocket.OPEN.
Basically, OPEN needs to be from the object itself instead of the WebSocket import.  Not sure why this is, but it seems to
be Typescript specific.  The issue this causes is the game will not work.  It produces a 500 error code.

The current plan is to merge this branch, get started on frontend testing, and improve the backend testing suite along with the necessary future additions.

### CODE REVIEW

File: backend/src/game.ts [hunk 1]
```diff
@@ -100,7 +100,7 @@ function endQuestion(gameId: GameId) {
 	};
 	const hostId = game.hostId;
 	const hostSocket = userSockets.get(hostId);
-	if (hostSocket !== undefined && hostSocket.readyState === WebSocket.OPEN) {
+	if (hostSocket !== undefined && hostSocket.readyState === hostSocket.OPEN) {
```
Comment (190n) [inline backend/src/game.ts]: Is this really correct/necessary? It seems like both WebSocket.OPEN and (some WebSocket instance).OPEN exist and are identical. IMO the one on the WebSocket class itself is clearer.

**Review by 190n — APPROVED** [review]
Looks great. Thanks for the documentation too.
